app/components/sidebars/main/channels_list/list/list.js

```diff
@@ -100,7 +101,7 @@ export default class List extends PureComponent {
             };
         case SidebarSectionTypes.PUBLIC:
             return {
-                action: this.goToMoreChannels,
+                action: canJoinPublicChannels ? this.goToMoreChannels : null,
```

Comment:
this will prevent the user to open the screen but the `+` icon will still appear in the UI, we should remove it from the UI as well.

Reply:
I don't think so, I think the action is what is checked to show the + icon. Maybe I'm wrong, I can double check it, but I tested it manually and the + sign wasn't shown.

Reply:
you are right, it does not render the action if action is not defined.